src/sources/generator.rs

```diff
@@ -18,25 +18,106 @@ use tokio::time::{interval, Duration};
 #[derive(Clone, Debug, Default, Deserialize, Serialize)]
 #[serde(deny_unknown_fields)]
 pub struct GeneratorConfig {
-    #[serde(default)]
-    sequence: bool,
-    lines: Vec<String>,
     #[serde(default)]
     batch_interval: Option<f64>,
     #[serde(default = "usize::max_value")]
     count: usize,
+    format: OutputFormat,
+}
+
+#[derive(Clone, Debug, Derivative, Deserialize, Serialize)]
+#[derivative(Default)]
+#[serde(rename_all = "snake_case")]
+pub enum OutputFormat {
```

Comment:
I suggest adding `#[serde(tag = "format")]` to this, which allows the following in configuration:
```toml
format = "round_robin"
sequence = true
items = ["line1", "line2"]
```
This general structure makes sense to me, though.

Reply:
Oh, nice. Yeah, I was struggling a bit with the test failures and am glad that the solution is so simple.